### ISSUE TITLE
Errors with R markdown scripts in WGBS pipeline

### ISSUE THREAD

**Issue by alexgreenlon** [maxplanck-ie/snakepipes#733]
Hello, I am running the snakePipes WGBS function on the test data. The pipeline runs almost to completion, but throws errors in the final steps, seemingly related to some of the r scripts. 

I acquired the test data with these commands:
```
cd /data
mkdir indices
wget https://zenodo.org/record/1475957/files/GRCm38_gencode_snakePipes.tgz?download=1
mv GRC* indices/GRCm38.tgz
cd indices
tar xf GRCm38.tgz
rm GRCm38.tgz
cd ..
# then editing the /data1/conda/envs/snakePipes/lib/python3.8/site-packages/snakePipes/shared/organisms/mm10.yaml
# to point to the GCRm38 files. 
```
Here is the command I used: 
```
WGBS --DAG --trim --local --DMRprograms metilene -j 16 --sampleSheet /data/sampleSheet.tsv -i /data/Habibi2013_chr6_F3A/fastq/  -o /data/test.local-samtools mm10
```

Here is the final portion of the log file that notes the errors:
```
70 of 73 steps (96%) done

[Mon Dec 14 20:23:39 2020]
rule metileneReport:
    input: metilene_sampleSheet_genome_minCoverage5/DMRs.txt, metilene_sampleSheet_genome_minCoverage5/metilene.IN.txt
    output: metilene_sampleSheet_genome_minCoverage5/Stats_report.html
    jobid: 21
    benchmark: metilene_sampleSheet_genome_minCoverage5/.benchmark/metileneReport.benchmark

Activating conda environment: /data1/conda/envs/f2e37f84292dd18eef9ef65979a59a6b
Loading required package: rhdf5
Quitting from lines 82-124 (tmpw8x61z_v.WGBS_metileneQC.Rmd) 
Error in makeTxDbFromGFF(gtfFile, format = "gtf") : 
  could not find function "makeTxDbFromGFF"
Calls: <Anonymous> ... handle -> withCallingHandlers -> withVisible -> eval -> eval
In addition: Warning message:
package 'tidyr' was built under R version 3.6.3 
Execution halted
[Mon Dec 14 20:23:44 2020]
Error in rule metileneReport:
    jobid: 21
    output: metilene_sampleSheet_genome_minCoverage5/Stats_report.html
    conda-env: /data1/conda/envs/f2e37f84292dd18eef9ef65979a59a6b

RuleException:
CalledProcessError in line 454 of /data1/conda/envs/snakePipes/lib/python3.8/site-packages/snakePipes/shared/rules/WGBS.snakefile:
Command 'source /data1/conda/bin/activate '/data1/conda/envs/f2e37f84292dd18eef9ef65979a59a6b'; set -euo pipefail;  Rscript --vanilla -e 'rmarkdown::render("/data/test.local-samtools/.snakemake/scripts/tmpw8x61z_v.WGBS_metileneQC.Rmd", output_file="/data/test.local-samtools/metilene_sampleSheet_genome_minCoverage5/Stats_report.html", quiet=TRUE, knit_root_dir = "/data/test.local-samtools", params = list(rmd="/data/test.local-samtools/.snakemake/scripts/tmpw8x61z_v.WGBS_metileneQC.Rmd"))'' returned non-zero exit status 1.
  File "/data1/conda/envs/snakePipes/lib/python3.8/site-packages/snakePipes/shared/rules/WGBS.snakefile", line 454, in __rule_metileneReport
  File "/data1/conda/envs/snakePipes/lib/python3.8/concurrent/futures/thread.py", line 57, in run
Job failed, going on with independent jobs.
Warning messages:
1: package 'pander' was built under R version 3.6.3 
2: package 'carData' was built under R version 3.6.3 
3: package 'matrixStats' was built under R version 3.6.3 
[Mon Dec 14 20:23:44 2020]
Finished job 7.
71 of 73 steps (97%) done
Exiting because a job execution failed. Look above for error message
Complete log: /data/test.local-samtools/.snakemake/log/2020-12-14T200549.864554.snakemake.log

 !!! ERROR in WGBS workflow! !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

The above was on an Amazon EC2 c4.4xlarge instance with 16 cores and 30 gb RAM running Ubuntu 20.04. I installed on my macbook pro running apple OS 10.14.6 with 4 cores and 32 gb RAM and got essentially the same error.

**Comment by katsikora** [comment]
Hi Alexgreenlon,

it looks like there is an issue with one of the functions in the GenomicRanges package. Either the conda environment didn't build successfully, or there is some function export issue. I'll have a look . On our side, WGBS is running through without errors, but we last  built our conda envs a few weeks ago. Maybe there is a temporary issue with some binaries.

**Comment by katsikora** [comment]
I can reproduce your error with a freshly build conda env on our system (CentOS7).

**Comment by katsikora** [comment]
Alright, this is fixed by adding an explicit export for GenomicFeatures function that's failing above to the Rmd file, updating R to 3.6.3 and updating dplyr to 1.0.2 in the wgbs.yaml. I'll add the fixes to the next release.

**Comment by alexgreenlon** [comment]
Great, thanks Katarzyna.

**Comment by katsikora** [comment]
The fix is now available in release 2.4.2.
